c10/test/core/SymInt_test.cpp

```diff
@@ -35,4 +36,167 @@ TEST(SymIntTest, Overflows) {
 }
 #endif
 
+namespace {
+
+class ConstantIntPretendingToBeSymbolicSymNodeImpl
```

Comment:
What's going on here?

Reply:
I needed a SymNodeImpl that would actually hit the `!maybe_as_int()` path in all the operators (and have working arithmetic) so I could get full test coverage. `ConstantSymNodeImpl` is `maybe_as_int()`. 

I'll add a code comment for this.